test/specs/default-controls.test.js

```diff
@@ -0,0 +1,371 @@
+    });
+
+    // testing extreme scenarios
+    it('should return true for slide index >= 4', () => {
+      const alignLeftIndex3 = {
+        wrapAround: false,
+        slidesToShow: 7,
+        currentSlide: 3,
+        cellAlign: 'left',
+        slideCount: 11
+      };
+      expect(instance.nextButtonDisabled(alignLeftIndex3)).toEqual(false);
+
+      const alignLeftIndex4 = {
+        wrapAround: false,
+        slidesToShow: 7,
+        currentSlide: 4,
+        cellAlign: 'left',
+        slideCount: 11
+      };
+      expect(instance.nextButtonDisabled(alignLeftIndex4)).toEqual(true);
+    });
+
+    // testing edge case scenarios
+    it('should return valid true or false value in edge cases', () => {
+      // when slideCount < slidesToShow and slideCount < slidesToShow
+      const alignLeftEdgeCase1 = {
+        wrapAround: false,
+        slidesToShow: 6,
+        currentSlide: 0,
+        cellAlign: 'left',
+        slideCount: 5
+      };
+      expect(instance.nextButtonDisabled(alignLeftEdgeCase1)).toEqual(true);
+
+      // when slideCount ===  slidesToShow
+      const alignLeftEdgeCase2 = {
+        wrapAround: false,
+        slidesToShow: 5,
+        currentSlide: 0,
+        cellAlign: 'left',
+        slideCount: 5
+      };
+      expect(instance.nextButtonDisabled(alignLeftEdgeCase2)).toEqual(true);
+    });
+  });
+
+  describe('#nextButtonDisable for cellAlign = `center`', () => {
+    let instance;
+
+    beforeEach(async () => {
+      const wrapper = mount(<NextButton buttonDisable={() => null} />);
+      instance = wrapper.instance();
+    });
+
+    // testing smaller number of pages
+    it('should return true for slide index 5', () => {
+      const alignCenterTest1 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 4,
+        cellAlign: 'center',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(alignCenterTest1)).toEqual(false);
+
+      const alignCenterTest2 = {
+        wrapAround: false,
+        slidesToShow: 3,
+        currentSlide: 5,
+        cellAlign: 'center',
+        slideCount: 6
+      };
+      expect(instance.nextButtonDisabled(alignCenterTest2)).toEqual(true);
+    });
+
+    // testing extreme scenarios
+    it('should return true for slide index 22', () => {
+      const alignCenterTest3 = {
+        wrapAround: false,
+        slidesToShow: 7,
+        currentSlide: 21,
+        cellAlign: 'center',
+        slideCount: 23
+      };
+      expect(instance.nextButtonDisabled(alignCenterTest3)).toEqual(false);
+
+      const alignCenterTest4 = {
+        wrapAround: false,
+        slidesToShow: 7,
+        currentSlide: 22,
+        cellAlign: 'center',
+        slideCount: 23
+      };
+      expect(instance.nextButtonDisabled(alignCenterTest4)).toEqual(true);
+    });
+
+    // testing edge case scenarios
+    it('should return true with slide index of 4 when slideCount < slidesToShow', () => {
```

Comment:
This is a great example of what I'm trying to convey in the comments above! This test statement is very clear